src/digest/base.cr

```diff
@@ -1,92 +1,143 @@
 require "base64"
 
+module Digest
+  class ApiMisuse < Exception
```

Comment:
```suggestion
  class DigestFinalizedError < Exception
```

Reply:
Somewhere I read having lots of exception classes was bad for compiler performance.

This may be the first exception but it's intended as a catch all for further misuse cases.  Would you still like it changed?

Reply:
> This may be the first exception but it's intended as a catch all for further misuse cases.

That's future-proofing, I'd better stick to the use-cases at hand, and solve the problems as they arise - there would be no problem with adding a common ancestor l8r on if deemed necessary - named probably more like `Digest::Error`.

> Would you still like it changed?

yes plz.